src/command-handler.ts

```diff
@@ -39,124 +35,97 @@ class CommandHandler {
-                } else {
-                    interaction.reply({ content: 'Currently not tracking any players', ephemeral: true });
-                }
-            } else if (interaction.commandName === 'check') {
-                CommandHandler.failIfDisabled();
-                const rsn = interaction.options.getString('username', true);
-                try {
-                    // Retrieve the player's hiscores data
-                    const data = await fetchHiScores(rsn);
-                    let messageText = '';
-                    // Create skills message text
-                    const totalLevel: string = (data.totalLevel ?? '???').toString();
-                    const baseLevel: string = (data.baseLevel ?? '???').toString();
-                    messageText += `${SKILLS_NO_OVERALL.map(skill => `**${data.levels[skill] ?? '?'}** ${skill}`).join('\n')}\n\nTotal **${totalLevel}**\nBase **${baseLevel}**`;
-                    // Create bosses message text if there are any bosses with one or more kills
-                    if (BOSSES.some(boss => data.bosses[boss])) {
-                        messageText += '\n\n' + BOSSES.filter(boss => data.bosses[boss]).map(boss => `**${data.bosses[boss]}** ${getBossName(boss)}`).join('\n');
-                    }
-                    // Create clues message text if there are any clues with a score of one or greater
-                    if (CLUES_NO_ALL.some(clue => data.clues[clue])) {
-                        messageText += '\n\n' + CLUES_NO_ALL.filter(clue => data.clues[clue]).map(clue => `**${data.clues[clue]}** ${clue}`).join('\n');
-                    }
-                    replyUpdateMessage(interaction, messageText, 'overall', {
-                        title: rsn,
-                        url: `${CONSTANTS.hiScoresUrlTemplate}${encodeURI(rsn)}`
-                    });
-                } catch (err) {
-                    if (err instanceof Error) {
-                        logger.log(`Error while fetching hiscores (check) for player ${rsn}: ${err.toString()}`);
-                        interaction.reply(`Couldn't fetch hiscores for player **${rsn}** :pensive:\n\`${err.toString()}\``);
-                    }
-                }
-            } else if (interaction.commandName === 'channel') {
+            if (command.failIfDisabled) {
                 CommandHandler.failIfDisabled();
+            }
+            if (command.privileged) {
                 CommandHandler.assertIsAdmin(interaction);
-                const guildId = CommandHandler.getInteractionGuildId(interaction);
-                await updateTrackingChannel(guildId, interaction.channelId);
-                const textChannel = interaction.channel as TextBasedChannel;
-                state.setTrackingChannel(guildId, textChannel);
-                await interaction.reply('Player experience updates will now be sent to this channel');
             }
+            command.execute && await command.execute(interaction);
```

Comment:
Can you make this a proper if-statement, and reply with a warning the else case?